File: fibonacci/fibo.py
import time

def fibonacci(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1
    else:
        return fibonacci(n-1) + fibonacci(n-2)

if __name__ == "__main__":
    for x in range(3):
        fibonacci(3)
    time.sleep(0)
    start = time.time()
    for x in range(0, 30):
        fibonacci(x)
    end = time.time()
    print end - start

File: fibonacci/fibo.java
import java.math.BigInteger;

public class fibo {
    static final BigInteger ZERO = new BigInteger("0");
    static final BigInteger ONE = new BigInteger("1");
    static final BigInteger TWO = new BigInteger("2");
    public static BigInteger fibonacci(BigInteger n) {
        if (n.equals(ZERO)) {
            return ZERO;
        } else if (n.equals(ONE)) {
            return ONE;
        }
        return fibonacci(n.subtract(ONE)).add(fibonacci(n.subtract(TWO)));
    }
    public static void main(String[] argv) {
        long start = System.currentTimeMillis();
        for (int i=0; i<30; i++) {
            System.out.println(fibonacci(new BigInteger(String.valueOf(i))));
        }
        long end = System.currentTimeMillis();
        System.out.println(1.0*(end - start)/1000);
    }
}
